libs/common-utils/src/getChainIdImmediately.ts

```diff
@@ -1,6 +1,11 @@
-import { JsonRpcProvider } from '@ethersproject/providers'
+import { JsonRpcProvider, Provider } from '@ethersproject/providers'
+
+export async function getChainIdImmediately(provider: JsonRpcProvider | Provider): Promise<number | undefined> {
+  if (!isJsonRpcProvider(provider)) {
+    console.error('Provider is not a JsonRpcProvider')
+    return undefined
```

Comment:
don't  you want to throw in this case?

Also, why we return undefined in the other catch?

Reply:
I'm a bit affraid of making it blocking.
If some provider is not JsonRpcProvider, then it won't be able to make transactions at all.

Reply:
Throwing or not it WILL block the user, because `undefined` is not a valid chainId.

If you guys think we should not block, we can skip the check when this call returns undefined.

Then we are back at no last minute check for those cases.

Reply:
<img width="534" alt="image" src="https://github.com/user-attachments/assets/6f4d9644-a6d0-4182-b5f6-2e639414fe32" />

Caught the error with Trust wallet via WalletConnect

Reply:
Trezor doesn't even get connected:
<img width="689" alt="image" src="https://github.com/user-attachments/assets/e6ceb8eb-a5dd-4349-9b24-6fc3edca81bd" />

But it works on prod.

Reply:
Fixed Trezor connecting:
https://github.com/cowprotocol/cowswap/pull/5357/commits/cf74448bd90b64a4df373fac05a9578a9a6f1f5e

Sorry @alfetopito I just pushed it in your branch

Reply:
+ Fix for transaction sending:
https://github.com/cowprotocol/cowswap/pull/5357/commits/42821e718332d57fd16e08754639f68225d3dc78

Reply:
Thank you!

There is another place that I changed that also needs fixing
![image](https://github.com/user-attachments/assets/ddc10b4b-d201-4b62-a93b-42f6e0866cbb)
https://github.com/cowprotocol/cowswap/pull/5354/files

Reply:
Trezor is good now.
The only problem is with WalletConnect:
https://github.com/cowprotocol/cowswap/pull/5357#discussion_r1935238912